src/Tizen.Nlp/Tizen.Nlp/NaturalLanguageProcess.cs

```diff
@@ -0,0 +1,279 @@
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Tizen.Applications;
+
+namespace Tizen.Nlp
+{
+
+    /// <summary>
+    /// This class contains the methods in the NLP processing.
+    /// </summary>
+    /// <since_tizen> 5 </since_tizen>
+    public class NaturalLanguageProcess
+    {
+        private Message _msg;
+        private readonly Message.NotifyCb _noti = new Message.NotifyCb();
+        private readonly string _tag;
+        private const string ServiceId = "org.tizen.nlp.service";
+        private delegate bool LangDetectCallback(MessageReceivedEventArgs e);
+        private delegate bool WordTokenizeCallback(MessageReceivedEventArgs e);
+        private delegate bool PostagCallback(MessageReceivedEventArgs e);
+        private delegate bool NamedEntityRecognitionCallback(MessageReceivedEventArgs e);
+        private delegate bool LemmatizeCallback(MessageReceivedEventArgs e);
+        private int _requestIdPos = 0;
+        private int _requestIdLang = 0;
+        private int _requestIdNeChunk = 0;
+        private int _requestIdWordTokenize = 0;
+        private int _requestIdLemmatize = 0;
+        private readonly Dictionary<int, PostagCallback> _mapsPosTag = new Dictionary<int, PostagCallback>();
+        private readonly Dictionary<int, LangDetectCallback> _mapsLangDetect = new Dictionary<int, LangDetectCallback>();
+        private readonly Dictionary<int, WordTokenizeCallback> _mapsWordTokenize = new Dictionary<int, WordTokenizeCallback>();
+        private readonly Dictionary<int, NamedEntityRecognitionCallback> _mapsNamedEntity = new Dictionary<int, NamedEntityRecognitionCallback>();
+        private readonly Dictionary<int, LemmatizeCallback> _mapsLemmatize = new Dictionary<int, LemmatizeCallback>();
+
+        private void MakeRequest(string cmd, string sentence, int requestid)
+        {
+            Bundle b = new Bundle();
+            b.AddItem("command", cmd);
+            b.AddItem("info", sentence);
+            b.AddItem("request_id", requestid.ToString());
+            _msg.Send(b);
+        }
+
+        private void ResultReceived(string sender, Bundle msg)
+        {
+            Log.Debug(_tag, "OnReceived ++");
+            MessageReceivedEventArgs e = new MessageReceivedEventArgs();
+            int requestid;
+            Dictionary<string, string[]> result = new Dictionary<string, string[]>();
```

Comment:
You can get commonly used item in this block, it can remove duplicated code. 
```c#
Dictionary<string, string[]> result = new Dictionary<string, string[]>();
result.Add("token", (string[])msg.GetItem("return_token"));
e.RequestId = requestid = int.Parse((string)msg.GetItem("request_id"));
```
```